spec/use_cases/certificate_exporter_spec.rb

```diff
@@ -0,0 +1,35 @@
+require "zip"
+
+describe UseCases::CertificateExporter do
+  let(:organisation) { create(:organisation, cba_enabled: true) }
+  let(:certficate_file_path) do
+    zipfile = Tempfile.new("out.zip")
+    zipfile.write(Gateways::S3.new(**Gateways::S3::CERTIFICATES).read)
+    zipfile.close
+    zipfile.path
+  end
+
+  it "zips up certificates" do
+    create_list(:certificate, 2, organisation:)
+
+    UseCases::CertificateExporter.export
+
+    Zip::File.open(certficate_file_path) do |files|
+      expect(files.count).to eq(2)
+      expect(files.map { |file| files.read(file) }).to match(Certificate.pluck(:content))
+    end
+  end
+
+  it "does not zip up certificates with children" do
+    root_key = OpenSSL::PKey::RSA.new(512)
+    create(:certificate, key: root_key, subject: "/CN=rootca", organisation:)
+    intermediate_ca = create(:certificate, issuing_key: root_key, issuing_subject: "/CN=rootca", organisation:)
+
+    UseCases::CertificateExporter.export
+
+    Zip::File.open(certficate_file_path) do |files|
+      expect(files.count).to eq(1)
+      expect(files.read(files.first)).to eq(intermediate_ca.content)
+    end
+  end
+end
```

Comment:
It think it would be good to have a test to handle s3 (AWS) exceptions, you can mock the s3 resource and make it return an exception, which is expected and caught, something like.
https://stackoverflow.com/questions/26665520/rspec-how-to-stub-a-3rd-party-exception